src/replication.c

```diff
@@ -1715,58 +1714,70 @@ void readSyncBulkPayload(connection *conn) {
     if (server.aof_state != AOF_OFF) stopAppendOnly();
 
-    /* When diskless RDB loading is used by replicas, it may be configured
-     * in order to save the current DB instead of throwing it away,
-     * so that we can restore it in case of failed transfer. */
-    if (use_diskless_load &&
-        server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB)
-    {
-        /* Create a backup of server.db[] and initialize to empty
-         * dictionaries. */
-        diskless_load_backup = disklessLoadMakeBackup();
-    }
+    if (use_diskless_load && server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB) {
+        /* Initialize empty tempDb dictionaries. */
+        diskless_load_tempDb = disklessLoadInitTempDb();
 
-    /* Replica starts to apply data from new master, we must discard the cached
-     * master structure. */
-    serverAssert(server.master == NULL);
-    replicationDiscardCachedMaster();
+        moduleFireServerEvent(REDISMODULE_EVENT_REPL_ASYNC_LOAD,
+                              REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_STARTED,
+                              NULL);
+    } else {
+        /* Replica starts to apply data from new master, we must discard the cached
+        * master structure. */
+        serverAssert(server.master == NULL);
+        replicationDiscardCachedMaster();
 
-    /* We want our slaves to resync with us as well, if we have any sub-slaves.
-     * The master already transferred us an entirely different data set and we
-     * have no way to incrementally feed our slaves after that. */
-    disconnectSlaves(); /* Force our slaves to resync with us as well. */
-    freeReplicationBacklog(); /* Don't allow our chained slaves to PSYNC. */
+        /* We want our slaves to resync with us as well, if we have any sub-slaves.
+        * The master already transferred us an entirely different data set and we
+        * have no way to incrementally feed our slaves after that. */
+        disconnectSlaves(); /* Force our slaves to resync with us as well. */
+        freeReplicationBacklog(); /* Don't allow our chained slaves to PSYNC. */
 
-    /* We call to emptyDb even in case of REPL_DISKLESS_LOAD_SWAPDB
-     * (Where disklessLoadMakeBackup left server.db empty) because we
-     * want to execute all the auxiliary logic of emptyDb (Namely,
-     * fire module events) */
-    emptyDb(-1,empty_db_flags,replicationEmptyDbCallback);
+        serverLog(LL_NOTICE, "MASTER <-> REPLICA sync: Flushing old data");
+        emptyDb(-1,empty_db_flags,replicationEmptyDbCallback);
+    }
 
     /* Before loading the DB into memory we need to delete the readable
-     * handler, otherwise it will get called recursively since
-     * rdbLoad() will call the event loop to process events from time to
-     * time for non blocking loading. */
+    * handler, otherwise it will get called recursively since
+    * rdbLoad() will call the event loop to process events from time to
+    * time for non blocking loading. */
```

Comment:
some of your multi-line block comments have indentation issues .
in this case it's not even a new comment, so i suppose maybe your editor is doing this?
please go over them.
```suggestion
     * handler, otherwise it will get called recursively since
     * rdbLoad() will call the event loop to process events from time to
     * time for non blocking loading. */
```